deepspeed/runtime/pipe/engine.py

```diff
@@ -588,14 +598,23 @@ def _exec_forward_pass(self, buffer_id):
 
         # Partition the outputs if we are not the last stage
         if self.is_pipe_partitioned and not self.is_last_stage():
-            assert torch.is_tensor(outputs)
-            part = PartitionedTensor(tensor=outputs,
+            if isinstance(outputs, tuple):
+                first_output = outputs[0]
+                # TODO: Improve pipe partitioning to pass multiple tensors that require grads
+                assert all([torch.is_tensor(elt) and elt.requires_grad is False for elt in outputs[1:]])
+                outputs_tail = outputs[1:]
+            elif torch.is_tensor(outputs):
+                first_output = outputs
+                outputs_tail = []
+            else:
+                raise NotImplementedError
```

Comment:
```suggestion
                raise ValueError("expecting a tensor or a tuple of tensors")
```

Let's be user-friendly